tests/integ/test_mxnet_train.py

```diff
@@ -72,6 +72,47 @@ def test_deploy_model(mxnet_training_job, sagemaker_session):
         predictor.predict(data)
 
 
+def test_deploy_model_with_update_endpoint(mxnet_training_job, sagemaker_session):
+    endpoint_name = 'test-mxnet-deploy-model-{}'.format(sagemaker_timestamp())
+
+    with timeout_and_delete_endpoint_by_name(endpoint_name, sagemaker_session):
+        desc = sagemaker_session.sagemaker_client.describe_training_job(TrainingJobName=mxnet_training_job)
+        model_data = desc['ModelArtifacts']['S3ModelArtifacts']
+        script_path = os.path.join(DATA_DIR, 'mxnet_mnist', 'mnist.py')
+        model = MXNetModel(model_data, 'SageMakerRole', entry_point=script_path,
+                           py_version=PYTHON_VERSION, sagemaker_session=sagemaker_session)
+        model.deploy(1, 'ml.t2.medium', endpoint_name=endpoint_name)
+        old_production_variants = sagemaker_session.describe_endpoint(EndpointName=endpoint_name)['ProductionVariants']
+
+        model.deploy(1, 'ml.m4.xlarge', update_endpoint=True, endpoint_name=endpoint_name)
+        new_production_variants = sagemaker_session.describe_endpoint(EndpointName=endpoint_name)['ProductionVariants']
+
+        assert new_production_variants['InstanceType'] == 'ml.m4.xlarge'
+        assert old_production_variants['VariantName'] != new_production_variants['VariantName']
```

Comment:
Let's assert that the `sagemaker_session.describe_endpoint(EndpointName=endpoint_name)['EndpointConfigName']` aren't the same.

Other than that, I think doing the assertion of the values such as: `assert new_production_variants['InstanceType'] == 'ml.m4.xlarge'` would be better then doing a comparison.